CMakeLists.txt

```diff
@@ -435,114 +484,91 @@ function(is_sdk_requested name result_var_name)
   endif()
 endfunction()
 
-# FIXME: separate the notions of SDKs used for compiler tools and target
-# binaries.
-if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
+# FIXME: the parameters we specify in SWIFT_SDKS are lacking architecture specifics,
+# so we need to hard-code it. For example, the SDK for Android is just 'ANDROID', 
+# which we assume below to be armv7. 
+# The iOS SDKs all have their architectures hardcoded because they are just specified by name (e.g. 'IOS' or 'WATCHOS').
+# We can't cross-compile the standard library for another linux architecture,
+# because the SDK list would just be 'LINUX' and we couldn't disambiguate it from the host.
+# 
+# To fix it, we would need to append the architecture to the SDKs,
+# for example: 'OSX-x86_64;IOS-armv7;...etc'.
+# We could easily do that - we have all of that information in build-script-impl.
+
+set(SWIFT_PRIMARY_VARIANT_SDK_default  "${SWIFT_HOST_VARIANT_SDK}")
+set(SWIFT_PRIMARY_VARIANT_ARCH_default "${SWIFT_HOST_VARIANT_ARCH}")
+
+if("${SWIFT_HOST_VARIANT_SDK}" STREQUAL "LINUX")
+  
   set(CMAKE_EXECUTABLE_FORMAT "ELF")
-
   set(SWIFT_HOST_VARIANT "linux" CACHE STRING
       "Deployment OS for Swift host tools (the compiler) [linux].")
 
-  set(SWIFT_HOST_VARIANT_SDK "LINUX")
-  set(SWIFT_PRIMARY_VARIANT_SDK_default "LINUX")
-
-  # FIXME: This will not work while trying to cross-compile.
-  if("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "x86_64")
-    set(SWIFT_HOST_VARIANT_ARCH "x86_64")
-    set(SWIFT_PRIMARY_VARIANT_ARCH_default "x86_64")
-
-    if("${SWIFT_ANDROID_NDK_PATH}" STREQUAL "")
-      set(swift_can_crosscompile_stdlib FALSE)
+  # Calculate the host triple
+  if("${SWIFT_HOST_TRIPLE}" STREQUAL "")
```

Comment:
It hasn't been calculated yet. We check if it's empty to support explicitly specifying it (for example, there are the generic `arm` and `armhf` triples - `arm-linux-gnueabi` and  `arm-linux-gnueabi**hf**`). The `SWIFT_HOST_VARIANT_ARCH` is still just `arm`, so we would need the build script, which knows which target we want, to build for a generic `arm` architecture but with the `armhf` triple.

I think that's right - not 100%, but the worst this check does is let you override the host triple.